fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/database/DatabaseSpecificSQLGenerator.java

```diff
@@ -168,4 +168,14 @@ public String currentSchema() {
             throw new IllegalStateException("Database type is not supported for current schema " + databaseTypeResolver.databaseType());
         }
     }
+
+    public String getBoolean(boolean condition) {
+        if (databaseTypeResolver.isMySQL()) {
```

Comment:
https://dev.mysql.com/doc/refman/5.7/en/boolean-literals.html @adamsaghy

Reply:
i have the feeling, i might learn something new today :)

Reply:
Well...the truth lies between.
select * from stretchy_report where self_service_user_report = '0'; -> Valid
select * from stretchy_report where self_service_user_report = '1'; -> Valid
select * from stretchy_report where self_service_user_report = 0; -> Valid
select * from stretchy_report where self_service_user_report = 1; -> Valid
select * from stretchy_report where self_service_user_report = false; -> Valid
select * from stretchy_report where self_service_user_report = true; -> Valid
select * from stretchy_report where self_service_user_report = 'false'; -> Valid, but by sheer luck (no error, but evaluate as false)
select * from stretchy_report where self_service_user_report = 'true'; -> Invalid, no error and evaluate it as false

Unfortunately, the stretchy report SQLs are using sometimes as a string this value.

Reply:
Right but the thing is, if they search for the String version of true/false/1/0 that's already indicating a very bad pattern to me which should be fixed, don't you think?

Reply:
I can accept that, but... it is not "worse" pattern like providing a boolean to the tinyInt (which is number)

Reply:
>  it is not "worse" pattern like providing a boolean to the tinyInt 

I don't agree with that. If the database engine provides the true/false literals and it'll translate them into 1/0 values, that's much better if you're really representing boolean values instead of using the numbers themselves. It's just semantically better.

Same for the booleans in Java. The underlying JVM represents them as 0/1 bits and I bet you don't want to use numbers there too. ;-)